Traceback (most recent call last):
  File "C:\Users\FJvEn\miniconda3\envs\jupyterbook\lib\site-packages\jupyter_core\utils\__init__.py", line 154, in wrapped
    asyncio.get_running_loop()
RuntimeError: no running event loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\FJvEn\miniconda3\envs\jupyterbook\lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "C:\Users\FJvEn\miniconda3\envs\jupyterbook\lib\site-packages\nbclient\client.py", line 1319, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "C:\Users\FJvEn\miniconda3\envs\jupyterbook\lib\site-packages\jupyter_core\utils\__init__.py", line 158, in wrapped
    return loop.run_until_complete(inner)
  File "C:\Users\FJvEn\miniconda3\envs\jupyterbook\lib\asyncio\base_events.py", line 649, in run_until_complete
    return future.result()
  File "C:\Users\FJvEn\miniconda3\envs\jupyterbook\lib\site-packages\nbclient\client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "C:\Users\FJvEn\miniconda3\envs\jupyterbook\lib\site-packages\nbclient\client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\Users\FJvEn\miniconda3\envs\jupyterbook\lib\site-packages\nbclient\client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import pandas as pd
import plotly.graph_objects as go
from scipy.stats import linregress
import numpy as np

# Data inladen
pad = "DB_Final/Gecombineerde_dataset__Politiek_belang__alcoholconsumptie_en_geluk.csv"
subset = pd.read_csv(pad)
subset.columns = subset.columns.str.strip()

# Regressie berekenen
slope, intercept, r_value, p_value, std_err = linregress(subset["Very important in life: Politics"], subset["Cantril ladder score"])

# X-waarden voor lijn
x_vals = np.linspace(subset["Very important in life: Politics"].min(), subset["Very important in life: Politics"].max(), 100)
y_pred = slope * x_vals + intercept

# Plot starten
fig = go.Figure()

# Simpele en catchy hovertekst
fig.add_trace(go.Scatter(
    x=subset["Very important in life: Politics"],
    y=subset["Cantril ladder score"],
    mode='markers',
    marker=dict(
        size=8,
        color='steelblue',
        line=dict(width=1, color='black'),
        opacity=0.8
    ),
    text=subset["Entity"],
    hovertemplate="<b>%{text}</b><br>Mensen die politiek belangrijk vinden: %{x:.1f}%<br>Geluk score: %{y:.2f} op 10<extra></extra>",
    showlegend=False
))

# Regressielijn toevoegen
fig.add_trace(go.Scatter(
    x=x_vals,
    y=y_pred,
    mode='lines',
    line=dict(color='red', dash='dash'),
    name='Regressielijn'
))

# Layout
fig.update_layout(
    title="Relatie tussen politiek belang en geluk per land",
    xaxis_title="Percentage mensen dat politiek 'heel belangrijk' vindt",
    yaxis_title="Gemiddelde geluksscore (schaal 0 tot 10)",
    template="simple_white",
    showlegend=False
)

fig.show()
fig.write_html("politiek_geluk_regressie.html")

------------------


[1;31m---------------------------------------------------------------------------[0m
[1;31mModuleNotFoundError[0m                       Traceback (most recent call last)
Cell [1;32mIn[1], line 1[0m
[1;32m----> 1[0m [38;5;28;01mimport[39;00m[38;5;250m [39m[38;5;21;01mpandas[39;00m[38;5;250m [39m[38;5;28;01mas[39;00m[38;5;250m [39m[38;5;21;01mpd[39;00m
[0;32m      2[0m [38;5;28;01mimport[39;00m[38;5;250m [39m[38;5;21;01mplotly[39;00m[38;5;21;01m.[39;00m[38;5;21;01mgraph_objects[39;00m[38;5;250m [39m[38;5;28;01mas[39;00m[38;5;250m [39m[38;5;21;01mgo[39;00m
[0;32m      3[0m [38;5;28;01mfrom[39;00m[38;5;250m [39m[38;5;21;01mscipy[39;00m[38;5;21;01m.[39;00m[38;5;21;01mstats[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m linregress

[1;31mModuleNotFoundError[0m: No module named 'pandas'

